localstack/services/events/target.py

```diff
@@ -17,12 +21,43 @@
     sqs_queue_url_for_arn,
 )
 from localstack.utils.aws.client_types import ServicePrincipal
+from localstack.utils.json import extract_jsonpath
 from localstack.utils.strings import to_bytes
 from localstack.utils.time import now_utc
 
 LOG = logging.getLogger(__name__)
 
 
+def filter_event_with_target_input_path(
+    input_path: TargetInputPath, event: PutEventsRequestEntry
+) -> PutEventsRequestEntry:
+    event = extract_jsonpath(event, input_path)
+    return event
+
+
+def transform_event_with_input_transformer(
+    input_transformer: InputTransformer, event: PutEventsRequestEntry
+) -> PutEventsRequestEntry:
+    """
+    Process the event with the input transformer of the target event,
+    by replacing the message with the populated InputTemplate.
+    docs.aws.amazon.com/eventbridge/latest/userguide/eb-transform-target-input.html
+    """
+    try:
+        input_paths = input_transformer["InputPathsMap"]
+        input_template = input_transformer["InputTemplate"]
+    except KeyError as e:
+        LOG.error("%s key does not exist in input_transformer.", e)
+        raise e
+    for key, path in input_paths.items():
+        value = extract_jsonpath(event, path)
+        if not value:
+            value = ""
+        input_template = input_template.replace(f"<{key}>", value)
+    event = re.sub('"', "", input_template)
+    return event
+
```

Comment:
input path and input transformers are not purely in scope, but helpful to get certain tests passed that validate event pattern matching, due to the small change I felt it is valid to add it in this PR